### ISSUE TITLE
About python code

### ISSUE THREAD

**Issue by Yijing-Lu** [seemoo-lab/RF-Veil#1]
I want to ask if there is a readme file about python code? I have errors when I run the code.

**Comment by Baeumla** [comment]
Hi, we have not provided a readme in the git repository. Can you please give more insight on the error messages you receive?

**Comment by Yijing-Lu** [comment]
> 嗨，我们没有在git存储库中提供自述文件。您能否进一步了解收到的错误消息？
 Hello, when I read the .csv file, the following error appears.
![image](https://user-images.githubusercontent.com/82701139/115259860-d490e800-a164-11eb-9be0-43fff580449a.png)

**Comment by Baeumla** [comment]
Thanks for you answer. Are you trying to read an own csv file or a csv file from this repository?

It might also help if you could provide some of your code in order to see how you use the functions defined in `extract_fingerprint.py`.

In the long term, I should think about adding more details in the Readme.

**Comment by Yijing-Lu** [comment]
Thanks! Currently, I just called the function. Both my test csv file and the csv file in the repository will report an error when the function is called.

Do I need to write the main function to run the code?

**Comment by Baeumla** [comment]
Yes you’ll need to have your own main and call „extract_measurements“ function with the corresponding parameters. But it seems like there is an encoding issue in your files. I have used UTF-8

**Comment by Yijing-Lu** [comment]
Thank you, but there is still a bit of doubt. Why do the fingerprints obtained from CSI in the function "extract_measurements" need to be averaged? I don't understand.

**Comment by Baeumla** [comment]
We calculated the means for plotting and most importantly to obtain only one fingerprint from a whole set of frames. You can leave the calculation of the average fingerprint out, which leaves you with a fingerprint for every frame.

**Comment by uperr** [comment]
你好，我想请问一下关于，privacy_attack/data文件夹下的02 privacy attack reference s468_sorted.csv文件，我通过extract_fingerprint.py作出的图让我很困惑，论文中也没有提及
![image](https://user-images.githubusercontent.com/58211055/118129870-4f30e880-b42f-11eb-9d42-ad1fab81afef.png)

**Comment by uperr** [comment]
Hello, I would like to ask about the 02 privacy attack reference s468_sorted.csv file in the privacy_attack/data folder. The picture I made through extract_fingerprint.py confuses me. It is not mentioned in the paper. This picture is an attacker. Recovered fingerprints?
![QQ截图20210513210938](https://user-images.githubusercontent.com/58211055/118130274-c5354f80-b42f-11eb-95f9-ef0b4419900f.png)

**Comment by Baeumla** [comment]
Hello @uperr, the high standard deviation is caused by outliers that we always have in the recorded data sets. Try to plot the fingerprints without averaging and you'll see them. The reason for those outliers is that the NI AFW can only filter on MAC addresses. This filter is applied on MAC-layer but we are using the data from PHY layer. Hence, we still get frames from devices that we do not want. Implementing a filter on PHY frames that woks in real-time (i.e. matching the MAC address filter with PHY frames) is actually left for future work.
